nx/lib/nx.ex

```diff
@@ -9167,6 +9167,84 @@ defmodule Nx do
     end
   end
 
+  @doc """
+  Finds the variance of a tensor.
+
+  ## Examples
+
+  iex> Nx.var(Nx.tensor([[1, 2], [3, 4]]))
+
+  #Nx.Tensor<
+  f32
+  1.25
+  >
+  """
+  @doc type: :element
+  @spec var(tensor :: Nx.Tensor.t()) :: Nx.Tensor.t()
+  def var(%Nx.Tensor{shape: shape} = tensor) do
+    total =
+      shape
+      |> Tuple.to_list()
+      |> Enum.reduce(fn val, acc -> acc * val end)
+
+    mean = mean(tensor)
+
+    tensor
+    |> subtract(mean)
+    |> power(2)
+    |> sum()
+    |> divide(total)
+  end
+
+  def var(_), do: raise(ArgumentError, "must be a tensor")
+
+  @doc """
+  Finds the standard deviation of a tensor.
+
+  ## Examples
+
+  iex> Nx.std(Nx.tensor([[1, 2], [3, 4]]))
+
+  #Nx.Tensor<
+  f32
+  1.1180340051651
+  >
+  """
+  @doc type: :element
+  @spec std(tensor :: Nx.Tensor.t()) :: Nx.Tensor.t()
+  def std(%Nx.Tensor{} = tensor) do
+    sqrt(var(tensor))
+  end
+
+  def std(_), do: raise(ArgumentError, "must be a tensor")
+
+  @doc """
+  Normalizes the tensor by using standard scale.
+
+  ## Examples
+
+  iex> Nx.normalize(Nx.tensor([[1, 2], [3, 4]]))
+
+  #Nx.Tensor<
+  f32[2][2]
+  [
+    [-1.3416407108306885, -0.4472135901451111],
+    [0.4472135901451111, 1.3416407108306885]
+  ]
+  >
+  """
+  @doc type: :element
+  @spec normalize(tensor :: Nx.Tensor.t()) :: Nx.Tensor.t()
+  def normalize(%Nx.Tensor{} = tensor) do
+    mean = mean(tensor)
+
+    tensor
+    |> Nx.subtract(mean)
+    |> Nx.divide(std(tensor))
```

Comment:
What about zero `std` value here?

Reply:
what do you mean by zero ?

Reply:
I believe the standard_deviation can return 0 (for example if all values are equal). In this case, this function will error. Are there other formulas to implement `standard_scale`?

Reply:
Look that std is 0 only and only if all the values are equal. In this case, there is no possibility to achieve unit variance, so the data should be left as-is. And actually, I've searched how it is calculated in Sci-kit learn. They based on this article http://www.cs.yale.edu/publications/techreports/tr222.pdf.

Reply:
But this problematic case is still handled particularly. What is more, they catch the cases when the std is non-zero, but values are almost constant for stability issues. https://github.com/scikit-learn/scikit-learn/blob/7e1e6d09bcc2eaeba98f7e737aac2ac782f0e5f1/sklearn/preprocessing/_data.py#L84

Reply:
Yeah, let's give up on the standard_scale/normalize for now then and revisit later.